### PR TITLE
Update code and doc comments about cert chains

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/Tests/provisioning/ProvisioningE2ETests.cs [hunk 1]
```diff
@@ -678,7 +678,6 @@ public async Task DPS_Registration_AmqpWs_X509_IndividualEnrollment_InvalidGloba
                 auth,
                 clientOptions);
 
-            using var cts = new CancellationTokenSource(PassingTimeoutMilliseconds);
```
Comment (drwill-ms) [inline e2e/Tests/provisioning/ProvisioningE2ETests.cs:681]: This gave us 1 big timeout. If a single operation failed due to timeout, we would not try again. I broke it up so we'd get multiple attempts, with each attempt being a reasonable duration.

File: iothub/device/src/Authentication/IotHubConnectionCredentials.cs [hunk 2]
```diff
@@ -132,6 +132,13 @@ public IotHubConnectionCredentials(string iotHubConnectionString)
         /// <summary>
         /// The full chain of certificates from the one used to sign the client certificate to the one uploaded to the service.
         /// </summary>
+        /// <remarks>
+        /// If supplied, the certificate chain will be verified and sent to the service in the request by adding them to the current user
```
Comment (drwill-ms) [inline iothub/device/src/Authentication/IotHubConnectionCredentials.cs:136]: User-facing doc comment to help users navigate how this is used and how they can opt-in or -out of it.

File: iothub/device/src/Authentication/IotHubConnectionCredentials.cs [hunk 3]
```diff
@@ -132,6 +132,13 @@ public IotHubConnectionCredentials(string iotHubConnectionString)
...
+        /// intermediate certificate store.
```
Comment (CIPop) [inline iothub/device/src/Authentication/IotHubConnectionCredentials.cs:137]: Nit: "Intermediate Certification Authorities" store.
Optionally we can also point to https://learn.microsoft.com/en-us/troubleshoot/developer/webapps/iis/www-authentication-authorization/configure-intermediate-certificates (or similar).

**Review by CIPop — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**